### DISCUSSION TITLE
How do I retrieve all InternalId and names in a list

### DISCUSSION THREAD

**Discussion by darren-kimberbell** [netsuitephp/netsuite-php#122] [Q&A]
I need to retrieve all of the values for items in a custom list.  How can I do that via this PHP library.  I can't seem to find any examples to do this, and everything I have tried has been a dismal failure. :-(  Any help or guidance would be greatly appreciated!

Thanks so much!
Darren Nay
darren@kimberbell.com

**Comment by searsandrew** [comment]
I just did it on a project if you're still looking for this.
I stored the Internal ID in an env variable and fetched it using a config. I am using Laravel Actions package, and I have the fetch as its action, and then I call the class with other actions or from my scheduler depending on what it's doing (keeping an online list up-to-date with NetSuite as the point of truth in my case).

```        
$service = new NetSuiteService();

$list = new GetListRequest();
$list->baseRef = new RecordRef();
$list->baseRef->type = 'customList';
$list->baseRef->internalId = $internalId;

$result = $service->getList($list);
if($result->readResponseList->status->isSuccess)
{
     return $result->readResponseList->readResponse[0]->record->customValueList->customValue;
}
return [];
```

**Reply by brianfreytag** [reply]
Is there a way to do this with a different `baseref` than `recordref` to search via `scriptId` so that you don't need to have environment variables?

Edit:

It is, indeed, possible:

```php
$list = new GetListRequest();
$list->baseRef = new CustomizationRef();
$list->baseRef->type = RecordType::customList;
$list->baseRef->scriptId = $scriptId;
```
